frontend/src/components/Header/Header.tsx

```diff
@@ -44,32 +49,70 @@ export function Header(): JSX.Element {
 
   const [open, setOpen] = useState(false);
   const [listItems, setListItems] = useState([
-    { link: "/", text: t("homepage") }
+    { link: "/", text: t("homepage") },
   ]);
 
-  useEffect(() => {
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLButtonElement>(
+    null,
+  );
+
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  React.useEffect(() => {
     const baseItems = [{ link: "/", text: t("homepage") }];
     const additionalItems = [
       { link: "/projects", text: t("myProjects") },
-      { link: "/courses", text: t("myCourses") }
+      { link: "/courses", text: t("myCourses") },
     ];
-    if (isLoggedIn()) {
+    if (me.role !== "UNKNOWN") {
       setListItems([...baseItems, ...additionalItems]);
-    }
-    else {
+    } else {
       setListItems(baseItems);
     }
-  }, [t]);
+  }, [me, t]);
 
   return (
     <Box sx={{ flexGrow: 1 }}>
       <AppBar position="sticky">
         <Toolbar disableGutters>
-          <IconButton edge="start" onClick={() => setOpen(!open)} sx={{ color: "white", marginLeft: 0 }}>
-            <MenuIcon style={{fontSize:"2rem"}} />
+          <IconButton
+            edge="start"
+            onClick={() => setOpen(!open)}
+            sx={{ color: "white", marginLeft: 0 }}
+          >
+            <MenuIcon style={{ fontSize: "2rem" }} />
           </IconButton>
-          <TitlePortal/>
-          <LoginButton></LoginButton>
+          <TitlePortal />
+          {me.role !== "UNKNOWN" && (
```

Comment:
Keep the login button in the header also, that way people can log in from `404` for example page also